src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -0,0 +1,1754 @@
+                    SpanHelpers.AdjustIndices(broadcastedRight.Rank - 1, 1, ref curIndex, broadcastedRight.Lengths);
+                }
+            }
+            return true;
+        }
+        #endregion
+
+        #region Stack
+        // REVIEW: NEEDS A DIFFERENT NAME?
+        // JUST AN OVERLOAD FOR CONCATENATE?
+        public static Tensor<T> Stack<T>(Tensor<T>[] input, int axis = 0)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            if (input.Length < 2)
+                throw new ArgumentException("Must provide at least 2 tensors to Stack.");
+            if (axis < 0)
+                axis = input.Rank - axis;
+
+            Tensor<T>[] outputs = new Tensor<T>[input.Length];
+            for (int i = 0; i < input.Length; i++)
+            {
+                outputs[i] = Tensor.Unsqueeze(input[0], axis);
+            }
+            return Tensor.Concatenate<T>(outputs, axis);
+        }
+        #endregion
+
+        #region Reshape
+        // REVIEW: SENTINAL VALUE? CONSTANT VALUE FOR -1 WILDCARD?
+        public static Tensor<T> Reshape<T>(this Tensor<T> input, params nint[] lengths)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool> => Reshape(input, lengths.AsSpan());
+
+        public static Tensor<T> Reshape<T>(this Tensor<T> input, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            var arrLengths = lengths.ToArray();
+            // Calculate wildcard info.
+            if (lengths.Contains(-1))
+            {
+                if (lengths.Count(-1) > 1)
+                    throw new ArgumentException("Provided dimensions can only include 1 wildcard.");
+                var tempTotal = input._linearLength;
+                for (int i = 0; i < lengths.Length; i++)
+                {
+                    if (lengths[i] != -1)
+                    {
+                        tempTotal /= lengths[i];
+                    }
+                }
+                arrLengths[lengths.IndexOf(-1)] = tempTotal;
+
+            }
+
+            var tempLinear = SpanHelpers.CalculateTotalLength(ref arrLengths);
+            if (tempLinear != input.LinearLength)
+                throw new ArgumentException("Provided dimensions are not valid for reshaping");
+            var strides = SpanHelpers.CalculateStrides(arrLengths.Length, arrLengths);
+            return new Tensor<T>(input._values, arrLengths, strides.ToArray(), input.IsPinned);
+        }
+
+        public static SpanND<T> Reshape<T>(this SpanND<T> input, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            var arrLengths = lengths.ToArray();
+            // Calculate wildcard info.
+            if (lengths.Contains(-1))
+            {
+                if (lengths.Count(-1) > 1)
+                    throw new ArgumentException("Provided dimensions can only include 1 wildcard.");
```

Comment:
All exception messages should be in the resx

Reply:
All have now been moved there.